File: blockchain_integration/pi_network/marketplaces/marketplace/SilkRoad/models/order.py
# models/order.py

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class Order(BaseModel):
    id: int
    user_id: int
    product_id: int
    quantity: int
    price: float
    status: str
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

File: sidra_chain_integration/dex-project/dex/entities/Order/Order.ts
import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';

@Entity()
export class Order {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  userId: string;

  @Column()
  symbol: string;

  @Column()
  side: string;

  @Column()
  amount: number;

  @Column()
  price: number;

  @Column()
  type: string;

  @Column()
  status: string;

  @Column()
  createdAt: Date;

  @Column()
  updatedAt: Date;
}
